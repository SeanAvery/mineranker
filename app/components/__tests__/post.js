import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'
import { StyleSheetTestUtils } from 'aphrodite'

import Post from './../post'
import PostImage from './../post-image'
import PostVideo from './../post-video'

describe('<Post {...props} />', () => {
  let defaultProps: Post

  beforeEach(() => {
    StyleSheetTestUtils.suppressStyleInjection()

    defaultProps = {
      _id: 'abc',
      description: 'React Native has come a long way since it was open-sourced in 2015. Fewer than two years later, itâ€™s being used not only in Facebook and Facebook Ads Manager, but also in many other companies, from Fortune 500 companies to hot new startups',
      downvote: 24,
      og_image_url: 'localhost/photo-url',
      source: 'new york times',
      title: 'React Native at Instagram',
      upvote: 142,
      url: 'https://example.com/',
    }
  })

  afterEach(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection()
  })

  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Post {...defaultProps} />, div)
  })

  it('renders correctly', () => {
    const tree = renderer.create(
      <Post {...defaultProps} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders image', () => {
    const post = mount(
      <Post
        { ...defaultProps }
      />
    )

    expect(post.contains(
      <PostImage
        photo={'localhost/photo-url'}
      />
    )).toBe(true)
  })

  // it('renders video', () => {
    // const dom = (
      // <Post
        // { ...defaultProps }
        // video_url={'localhost/video-url'}
        // isMobile={true}
      // />
    // )
    // const post = mount(dom)

    // const children = post.find(PostVideo)

    // expect(children.length).toBe(1)
    // expect(children.first().props().photo).toEqual('localhost/photo-url')
    // expect(children.first().props().video).toEqual('localhost/video-url')
  // })

  it('renders children', () => {
    const child = (<div>Test 1</div>)
    const child2 = (<div>Test 2</div>)
    const dom = (
      <Post { ...defaultProps }>
        {child}
        {child2}
      </Post>
    )
    const post = mount(dom)

    expect(post.children().length).toBe(5)
    expect(post.children().contains(child)).toBe(true)
    expect(post.children().contains(child2)).toBe(true)

    // snapshot
    const tree = renderer.create(dom).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
