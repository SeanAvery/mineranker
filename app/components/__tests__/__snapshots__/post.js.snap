// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Post {...props} /> renders children 1`] = `
<div
  className="container_1ox283w"
>
  <div
    className="container_1ssoezr"
  >
    <img
      className="image_kt1q8t"
      src="localhost/photo-url"
    />
  </div>
  <div
    className="body_gexgkn"
  >
    <span
      className="title_njp6s6"
    >
      React Native at Instagram
    </span>
    <span
      className="description_14rca9u"
    >
      React Native has come a long way since it was open-sourced in 2015. Fewer than two years later, it’s being used not only in Facebook and Facebook Ads Manager, but also in many other companies, from Fortune 500 companies to hot new startups
    </span>
  </div>
  <div>
    Test 1
  </div>
  <div>
    Test 2
  </div>
  <div
    className="footer_diuw3w"
  >
    <div>
      <a
        className="vote_12qtyj8"
        href="javascript:void(0)"
        onClick={[Function]}
      >
        <img
          className="svg_1qn01rn"
          src="test-file-stub"
        />
        <span
          className="voteText_vxbm49"
        >
          142
        </span>
      </a>
      <a
        className="vote_12qtyj8"
        href="javascript:void(0)"
        onClick={[Function]}
      >
        <img
          className="svg_1qn01rn"
          src="test-file-stub"
        />
        <span
          className="voteText_vxbm49"
        >
          24
        </span>
      </a>
    </div>
    <div
      className="link_1e37oh4"
    >
      <a
        className="goto_12qtyj8"
        href="https://example.com/"
        target="_blank"
      >
        <span
          className="gotoText_vxbm49"
        >
          go to site
        </span>
        <img
          className="svg_1qn01rn"
          src="test-file-stub"
        />
      </a>
    </div>
  </div>
</div>
`;

exports[`<Post {...props} /> renders correctly 1`] = `
<div
  className="container_1ox283w"
>
  <div
    className="container_1ssoezr"
  >
    <img
      className="image_kt1q8t"
      src="localhost/photo-url"
    />
  </div>
  <div
    className="body_gexgkn"
  >
    <span
      className="title_njp6s6"
    >
      React Native at Instagram
    </span>
    <span
      className="description_14rca9u"
    >
      React Native has come a long way since it was open-sourced in 2015. Fewer than two years later, it’s being used not only in Facebook and Facebook Ads Manager, but also in many other companies, from Fortune 500 companies to hot new startups
    </span>
  </div>
  <div
    className="footer_diuw3w"
  >
    <div>
      <a
        className="vote_12qtyj8"
        href="javascript:void(0)"
        onClick={[Function]}
      >
        <img
          className="svg_1qn01rn"
          src="test-file-stub"
        />
        <span
          className="voteText_vxbm49"
        >
          142
        </span>
      </a>
      <a
        className="vote_12qtyj8"
        href="javascript:void(0)"
        onClick={[Function]}
      >
        <img
          className="svg_1qn01rn"
          src="test-file-stub"
        />
        <span
          className="voteText_vxbm49"
        >
          24
        </span>
      </a>
    </div>
    <div
      className="link_1e37oh4"
    >
      <a
        className="goto_12qtyj8"
        href="https://example.com/"
        target="_blank"
      >
        <span
          className="gotoText_vxbm49"
        >
          go to site
        </span>
        <img
          className="svg_1qn01rn"
          src="test-file-stub"
        />
      </a>
    </div>
  </div>
</div>
`;
